matching-notification-api/src/main/java/org/phenotips/matchingnotification/finder/internal/DefaultMatchFinderManager.java

```diff
@@ -60,30 +60,41 @@
     public void findMatchesForAllPatients(Set<String> serverIds, boolean onlyCheckPatientsUpdatedAfterLastRun)
     {
         this.logger.error("Finding matches for all patients for servers [{}] (only updated patients: [{}])",
-                serverIds.toString(), onlyCheckPatientsUpdatedAfterLastRun);
+            serverIds.toString(), onlyCheckPatientsUpdatedAfterLastRun);
 
-        List<String> patients = this.getPatientsList();
+        List<String> patients = this.getPatientsList(onlyCheckPatientsUpdatedAfterLastRun);
 
         for (MatchFinder matchFinder : this.matchFinderProvider.get()) {
             this.logger.debug("Starting matchFinder with priority = [{}]", matchFinder.getPriority());
-            matchFinder.findMatches(patients, serverIds, onlyCheckPatientsUpdatedAfterLastRun);
+            matchFinder.findMatches(patients, serverIds);
         }
     }
 
     /**
      * Returns a list of patient Ids.
      */
-    private List<String> getPatientsList()
+    private List<String> getPatientsList(boolean onlyCheckPatientsUpdatedAfterLastRun)
     {
         List<String> patients = new LinkedList<>();
 
         try {
             Query q = this.qm.createQuery(
                 "select doc.name "
                     + "from Document doc, "
-                    + "doc.object(PhenoTips.PatientClass) as patient "
-                    + "where patient.identifier is not null order by patient.identifier desc",
+                    + "doc.object(PhenoTips.PatientClass) as patient, "
+                    + "doc.object(PhenoTips.VisibilityClass) as visibility "
+                    + "where patient.identifier is not null and "
+                    // exclude solved patients
+                    + "patient.solved <> '1' and "
+                    // filter by last update date
+                    + "doc.contentUpdateDate > :startTime and "
+                    + "order by patient.identifier desc",
                 Query.XWQL);
+            if (onlyCheckPatientsUpdatedAfterLastRun) {
+                q.bindValue("startTime", System.currentTimeMillis());
```

Comment:
There is a problem here. `onlyCheckPatientsUpdatedAfterLastRun` means not patient updated after `System.currentTimeMillis()`, but updated after that particular matcher was run, which only that matcher knows (see line https://github.com/phenotips/patient-network/pull/198/commits/a31b758a238635fbc477b0894d0210d0506a4f71#diff-9a72985fda0ed745264848cb16cafa2fR123)

So the "is patient updated since last time" check should be done within the matcher and not here, sinc eit is matcher dependant

Reply:
Since last update info is stored in a separate document in phenomecentral space it does not depend on a matcher. Also we do not have any manager public method to use to get last update time, so it can be done here in the match finder manager. I will update PR.